src/Datasource/RepositoryInterface.php

```diff
@@ -115,6 +115,50 @@ public function updateAll($fields, $conditions);
     public function deleteAll($conditions);
 
     /**
+     * Deletes records individually by reading batches that match the conditions. For each
```

Comment:
Maybe now that it allows to save batches it should be called batchDelete or deleteBatch and then afterDeleteBatch and and afterDeleteBatchCommit etc... ?!

On the other hand it could also say:
`Deletes records that match the conditions individually. For each matched record Table::delete() is executed providing support for behaviors and events. You can specify a batch size to delete at once to improve performance on large amounts of records.` ... or similar